proto/src/control.rs

```diff
@@ -418,6 +470,44 @@ impl TryFrom<StructureTag> for LdapControl {
 
                 Ok(LdapControl::ManageDsaIT { criticality })
             }
+            "1.2.840.113556.1.4.474" => {
+                let value = value_tag
+                    .and_then(|t| t.match_class(TagClass::Universal))
+                    .and_then(|t| t.match_id(Types::OctetString as u64))
+                    .and_then(|t| t.expect_primitive())
+                    .ok_or(LdapProtoError::ControlBer)?;
+
+                let mut parser = Parser::new();
+                let (_, tag) = parser
+                    .parse(&value)
+                    .map_err(|_| LdapProtoError::ControlBer)?;
+
+                let mut tags = tag
+                    .match_class(TagClass::Universal)
+                    .and_then(|t| t.match_id(Types::Sequence as u64))
+                    .and_then(|t| t.expect_constructed())
+                    .ok_or(LdapProtoError::ControlBer)?;
+
+                let enum_tag = tags.pop().ok_or(LdapProtoError::ControlBer)?;
+
+                let value = enum_tag
+                    .match_class(TagClass::Universal)
+                    .and_then(|t| t.match_id(Types::Enumerated as u64))
+                    .and_then(|t| t.expect_primitive())
+                    .ok_or(LdapProtoError::ControlBer)?;
+
+                let code = value.first().ok_or(LdapProtoError::ControlBer)?;
+
+                Ok(LdapControl::ServerSortResult {
+                    sort_result: ServerSortResult {
+                        result_code: code
+                            .clone()
+                            .try_into()
+                            .map_err(|_| LdapProtoError::ControlBer)?,
+                        attribute_type: None, // TODO!
+                    },
+                })

```

Comment:
Nitpick: we typically don’t call `clone` on `Copy` types (unless writing generic code)

Suggestion:
```diff
@@ -496,14 +496,15 @@ impl TryFrom<StructureTag> for LdapControl {
                     .and_then(|t| t.expect_primitive())
                     .ok_or(LdapProtoError::ControlBer)?;

-                let code = value.first().ok_or(LdapProtoError::ControlBer)?;
+                let result_code = value
+                    .first()
+                    .copied()
+                    .and_then(|code| ServerSortResultCode::try_from(code).ok())
+                    .ok_or(LdapProtoError::ControlBer)?;

                 Ok(LdapControl::ServerSortResult {
                     sort_result: ServerSortResult {
-                        result_code: code
-                            .clone()
-                            .try_into()
-                            .map_err(|_| LdapProtoError::ControlBer)?,
+                        result_code,
                         attribute_type: None, // TODO!
                     },
                 })
```

Reply:
clippy complains if you do .clone() on a copy type too.

Reply:
updated